synapseclient/client.py

```diff
@@ -5247,3 +5270,146 @@ def _return_rest_body(self, response):
+        can_enable_inheritance (Boolean): (Read Only) Can the user delete the entity's access control list (so it inherits settings from an ancestor)?
+        owner_principal_id (Integer): (Read Only) The principal ID of the entity's owner (i.e. the entity's 'createdBy')
+        can_public_read (Boolean): (Read Only) Is this entity considered public?
+        can_moderate (Boolean): Can the user moderate the forum associated with this entity? Note that only project entity has forum.
+        is_certification_required (Boolean): (Read Only) Is the certification requirement enabled for the project of the entity?
+        is_entity_open_data (Boolean): (Read Only) Returns true if the Entity's DateType equals 'OPEN_DATA', indicating that the data is safe to be released to the public.
+    """
+
+    can_view: bool = None
+    """Can the user view this entity?"""
+    can_edit: bool = None
+    """Can the user edit this entity?"""
+    can_move: bool = None
+    """(Read Only) Can the user move this entity by changing its parentId?"""
+    can_add_child: bool = None
+    """Can the user add a child entity to this entity?"""
+    can_certified_user_edit: bool = None
+    """(Read Only) Can the user edit this entity once they become a Certified User?"""
+    can_certified_user_add_child: bool = None
+    """(Read Only) Can the user add a child entity to this entity once they become a Certified User?"""
+    is_certified_user: bool = None
+    """(Read Only) True, if the user has passed the user certification quiz."""
+    can_change_permissions: bool = None
+    """Can the user change the permissions of this entity?"""
+    can_change_settings: bool = None
+    """Can the user change the settings of this entity?"""
+    can_delete: bool = None
+    """Can the user delete this entity?"""
+    can_download: bool = None
+    """Are there any access requirements precluding the user from downloading this entity?"""
+    can_upload: bool = None
+    """(Read Only) Are there any access requirements precluding the user from uploading into this entity (folder or project)?"""
+    can_enable_inheritance: bool = None
+    """(Read Only) Can the user delete the entity's access control list (so it inherits settings from an ancestor)?"""
+    owner_principal_id: int = None
+    """(Read Only) The principal ID of the entity's owner (i.e. the entity's 'createdBy')"""
+    can_public_read: bool = None
+    """(Read Only) Is this entity considered public?"""
+    can_moderate: bool = None
+    """Can the user moderate the forum associated with this entity? Note that only project entity has forum."""
+    is_certification_required: bool = None
+    """(Read Only) Is the certification requirement enabled for the project of the entity?"""
+    is_entity_open_data: bool = None
+    """(Read Only) Returns true if the Entity's DateType equals 'OPEN_DATA', indicating that the data is safe to be released to the public."""
+
+    def __init__(self, data: dict):
```

Comment:
I'm not sure this is how we should create an instance of this class. The reason is that if you are overriding the init dunder method than it prevent you from creating a Permissions instance like:
```
my_other_permissions = Permissions(can_view=True)
print(my_other_permissions)
```

This might make more sense:
```
    @classmethod
    def from_dict(cls, data: dict) -> "Permissions":
        """
            Convert a data dictory to an instance of this dataclass.
        """
        return cls(
            can_view=data["canView"],
            can_edit=data["canEdit"],
            can_move=data["canMove"],
            can_add_child=data["canAddChild"],
            can_certified_user_edit=data["canCertifiedUserEdit"],
            can_certified_user_add_child=data["canCertifiedUserAddChild"],
            is_certified_user=data["isCertifiedUser"],
            can_change_permissions=data["canChangePermissions"],
            can_change_settings=data["canChangeSettings"],
            can_delete=data["canDelete"],
            can_download=data["canDownload"],
            can_upload=data["canUpload"],
            can_enable_inheritance=data["canEnableInheritance"],
            owner_principal_id=data["ownerPrincipalId"],
            can_public_read=data["canPublicRead"],
            can_moderate=data["canModerate"],
            is_certification_required=data["isCertificationRequired"],
            is_entity_open_data=data["isEntityOpenData"],
        )
```

In the `get_permissions` function:
`return Permissions.from_dict(data)`


Thoughts?